### ISSUE TITLE
Selection doesn't behave as expected. Selection color cannot be changed

### ISSUE THREAD

**Issue by frebib** [jwilm/alacritty#1343]
Linux X11, using `scrollback` @ https://github.com/jwilm/alacritty/commit/ad83dc63ac4a0c84d2092267521a86b1d697ef7f

Various issues I have noticed with 'selection' with the mouse:
- Mouse selection color is not configurable, [comparison with alacritty (top) and termite (bottom)](https://i.imgur.com/Fs4Tiqj.png)
  - Additionally foreground/background of highlighted text renders differently with alacritty but I feel like this should be discussed
- There is no "block selection" mode, as [demonstrated here in termite](https://i.imgur.com/jiJBMZn.png) by holding Ctrl whilst selecting
- Selection isn't 'deselected' when a selection in another window is made (I presume ignoring an X event?). This doesn't follow the behaviour that other TEs such as urxvt and termite use.

**Comment by chrisduerr** [comment]
Block selection is tracked in #978 but I'm not sure if it's going to be implemented anytime soon because the current way selection is represented in the code does not allow for block selections but only linear selections (directly from start point to the end including everything in between).

Selection color is currently just the inverse of foreground and background color, but having the option to freely select the color the selection background should be seems like a reasonable choice. In your termite screenshot it seems like the foreground color stays unchanged, I'm not sure if that's the best approach because it might lead to unreadable text.

I don't think removing selections when selection is made in another window has ever come up, but if there's an API for it I think it might be nice to include it. There might be some workflows that would get broken by this, so some more input on this would probably be good.

**Comment by nixpulvis** [comment]
If we ever build out real "modes" I'd love a mouseless selection interface (with vi bindings). But that's a rather complex addition, I could see arguments being made to not do it for the sake of simplicity.

**Comment by chrisduerr** [comment]
Yeah @nixpulvis I have similar feelings about both wanting it and understanding that it's probably too complex, but I believe there's an issue for tracking this already.
